flycheck.el

```diff
@@ -10077,6 +10078,31 @@ See URL `https://github.com/uber/prototool'."
   :enabled flycheck-prototool-project-root
   :predicate flycheck-buffer-saved-p)
 
+(defun flycheck-psalm-project-root (&optional _checker)
+  "Return the nearest directory with psalm.xml configuration."
+  (and buffer-file-name
+       (locate-dominating-file buffer-file-name "psalm.xml")))
+
+(flycheck-def-option-var flycheck-psalm-threads nil psalm
+  "If greater than one, Psalm will run analysis on multiple
+threads, speeding things up."
+  :type '(integer :tag "Number of threads in use")
+  :safe #'integerp)
+
+(flycheck-define-checker psalm
+  "Flycheck for Psalm, the free software static tool for PHP.
+See URL `https://psalm.dev/'."
+  :command ("psalm.phar " "--no-progress" "--output-format=emacs"
```

Comment:
This command is not the best specification.
 
[Installation - Documentation](https://psalm.dev/docs/running_psalm/installation/) describes how to install Psalm on each project basis.  We can never assume that the file `psalm.phar` is in `PATH`.

Reply:
`phpstan.el` takes a slightly complicated approach to solving that problem and Docker support.

https://github.com/emacs-php/phpstan.el/blob/master/phpstan.el#L262-L286